src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/RemoveAzureRmMlOpCluster.cs

```diff
@@ -58,26 +58,47 @@ public class RemoveAzureRmMlOpCluster : MachineLearningComputeCmdletBase
             HelpMessage = ResourceIdParameterHelpMessage)]
         public string ResourceId { get; set; }
 
+        [Parameter(ParameterSetName = ObjectParameterSet,
+            Mandatory = false, 
+            HelpMessage = IncludeAllResourcesParameterHelpMessage)]
+        [Parameter(ParameterSetName = ResourceIdParameterSet,
+            Mandatory = false, 
+            HelpMessage = IncludeAllResourcesParameterHelpMessage)]
+        [Parameter(ParameterSetName = CmdletParametersParameterSet,
+            Mandatory = false, 
+            HelpMessage = IncludeAllResourcesParameterHelpMessage)]
+        public SwitchParameter IncludeAllResources { get; set; }
+
         public override void ExecuteCmdlet()
         {
-            if (ShouldProcess(this.Name, @"Deleting operationalization cluster..."))
+            if (string.Equals(this.ParameterSetName, ObjectParameterSet, StringComparison.OrdinalIgnoreCase))
             {
-                if (string.Equals(this.ParameterSetName, ObjectParameterSet, StringComparison.OrdinalIgnoreCase))
-                {
-                    var resourceInfo = new ResourceIdentifier(InputObject.Id);
-                    ResourceGroupName = resourceInfo.ResourceGroupName;
-                    Name = resourceInfo.ResourceName;
-                }
-                else if (string.Equals(this.ParameterSetName, ResourceIdParameterSet, StringComparison.OrdinalIgnoreCase))
-                {
-                    var resourceInfo = new ResourceIdentifier(ResourceId);
-                    ResourceGroupName = resourceInfo.ResourceGroupName;
-                    Name = resourceInfo.ResourceName;
-                }
+                var resourceInfo = new ResourceIdentifier(InputObject.Id);
+                ResourceGroupName = resourceInfo.ResourceGroupName;
+                Name = resourceInfo.ResourceName;
+            }
+            else if (string.Equals(this.ParameterSetName, ResourceIdParameterSet, StringComparison.OrdinalIgnoreCase))
+            {
+                var resourceInfo = new ResourceIdentifier(ResourceId);
+                ResourceGroupName = resourceInfo.ResourceGroupName;
+                Name = resourceInfo.ResourceName;
+            }
+
+            var shouldProcessMessage = @"Deleting operationalization cluster";
+
+            if (IncludeAllResources.IsPresent)
+            {
+                var clusterToDelete = MachineLearningComputeManagementClient.OperationalizationClusters.Get(ResourceGroupName, Name);
+                var managedByResourceGroup = new ResourceIdentifier(clusterToDelete.ContainerRegistry.ResourceId).ResourceGroupName;
+
+                shouldProcessMessage += $" and supporting resource group {managedByResourceGroup}";
```

Comment:
Is the supporting resource group the only additional resource that will be deleted when -IncludeAllResources is specified?

Reply:
Yes it deletes that cluster and the supporting resource group only. The supporting resource group has a bunch of resources in it, I could change it to list them all out but it will include things like network security groups, nics, vms, disks, etc. It seemed too verbose to list all those resources individually so I think just warning that that resource group will be deleted should be enough.

Reply:
I see, I agree that this message would make the most sense.  Maybe it would be useful to also add in the message that all resources in the supporting resource group will be deleted.

Reply:
Good suggestion.